src/rdma/BaseRDMA.cc

```diff
@@ -77,38 +77,59 @@ void BaseRDMA::createBuffer() {
   }
 
   bool found = false;
-  //Choose rdma device on the correct numa node
+  //Choose rdma device based on the name or based on the correct numa node
   for (int i = 0; i < num_devices; i++)
   {
     ifstream numa_node_file;
-    numa_node_file.open(std::string(dev_list[i]->ibdev_path)+"/device/numa_node");
+    numa_node_file.open(std::string(dev_list[i]->ibdev_path) + "/device/numa_node");
     int numa_node = -1;
-    numa_node_file >> numa_node;
-    if (numa_node == m_numaNode)
+    if (numa_node_file) {
+      numa_node_file >> numa_node;
+    }
+    if(Config::RDMA_DEV_NAME.size() > 0 && Config::RDMA_DEV_NAME == std::string(dev_list[i]->name)) {
+      // this device could be selected by the config
+      if (numa_node != -1 && numa_node != m_numaNode) {
+        Logging::warn("Device was selected even though numa_node is not the right one (device has numa_node " + std::to_string(numa_node) + ", you selected " + std::to_string(m_numaNode) + ")");
+      }
+      ib_dev = dev_list[i];
+      found = true;
+      break;
+    }
+    else if (numa_node != -1 && numa_node == m_numaNode)
```

Comment:
e.g. RDMA_DEV_NAME = mlx5_1 and dev_list[i].name = mlx5_0 with both on the same numa region, the first if will evaluate correctly to false, but the else if will evaluate to true, since they are on the same region. 
Should test all devices for the right one string based first and only after that for the numa region.

another idea would be to check for the RDMA_INTERFACE, since this will be checked later as well.

Reply:
fixed in 108cf63dcb32e633c1e56fa050ce3ad05ff7fd85